### PR TITLE
DOC: Fixed formatting and errors in whatsnew v1.1.0

### PR DESCRIPTION
**Pull request by rhshadrach**
cc @TomAugspurger 

Minor fixes for formatting, spelling, and grammar. Please let me know if any of this is too nit-picky, I can revert to just the definite spelling/syntax errors.

### CODE REVIEW

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -1181,23 +1180,23 @@ ExtensionArray
 ^^^^^^^^^^^^^^
 
 - Fixed bug where :meth:`Series.value_counts` would raise on empty input of ``Int64`` dtype (:issue:`33317`)
-- Fixed bug in :func:`concat` when concatenating DataFrames with non-overlaping columns resulting in object-dtype columns rather than preserving the extension dtype (:issue:`27692`, :issue:`33027`)
+- Fixed bug in :func:`concat` when concatenating :class:`DataFrame`s with non-overlapping columns resulting in object-dtype columns rather than preserving the extension dtype (:issue:`27692`, :issue:`33027`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.1.0.rst:1183]: unwanted pattern ci failure - backticks incorrectly rendering because of missing spaces

Comment (rhshadrach) [reply doc/source/whatsnew/v1.1.0.rst]: Thanks - passing now.

**Comment by rhshadrach** [comment]
Failure is unrelated.

**Comment by TomAugspurger** [comment]
Reported upstream in https://github.com/numpy/numpy/issues/16939.

Thanks @rhshadrach.

**Pull request merged**